app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala

```diff
@@ -186,6 +197,83 @@ class RepresenteeController @Inject() (
     }
   }
 
+  def confirmPerson(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withCapacitorOrPersonalRepresentativeAnswers(request) { (representativeType, journey, answers) =>
+      val backLink = routes.RepresenteeController.enterId()
+
+      answers match {
+        case incompleteAnswers @ IncompleteRepresenteeAnswers(Some(_), Some(_), _, None, false, false) =>
+          Ok(confirmPersonPage(incompleteAnswers, representativeType, journey.isRight, confirmPersonForm, backLink))
+
+        case _ => Redirect(routes.RepresenteeController.checkYourAnswers())
+      }
+    }
+  }
+
+  def confirmPersonSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withCapacitorOrPersonalRepresentativeAnswers(request) { (representativeType, journey, answers) =>
+        val backLink = routes.RepresenteeController.enterId()
+
+        def handleForm(incompleteRepresenteeAnswers: IncompleteRepresenteeAnswers): Future[Result] =
+          confirmPersonForm
+            .bindFromRequest()
+            .fold(
+              formWithErrors =>
+                BadRequest(
+                  confirmPersonPage(
+                    incompleteRepresenteeAnswers,
+                    representativeType,
+                    journey.isRight,
+                    formWithErrors,
+                    backLink
+                  )
+                ),
+              isYes =>
+                if (isYes) {
+                  updateDraftReturnAndSession(
+                    incompleteRepresenteeAnswers.copy(hasConfirmedPerson = true),
+                    journey
+                  ).fold({ e =>
+                    logger.warn("Could not update draft return", e)
+                    errorHandler.errorResult()
+                  }, _ => Redirect(routes.RepresenteeController.checkYourAnswers()))
+                } else {
+                  updateDraftReturnAndSession(
+                    IncompleteRepresenteeAnswers.empty,
+                    journey
+                  ).fold({ e =>
+                    logger.warn("Could not update draft return", e)
+                    errorHandler.errorResult()
+                  }, _ => Redirect(routes.RepresenteeController.checkYourAnswers()))
```

Comment:
there's some duplication you can avoid here:
```scala
val newAnswers =
  if(isYes) incompleteRepresenteeAnswers.copy(hasConfirmedPerson = true)
  else IncompleteRepresenteeAnswers.empty

  updateDraftReturnAndSession(
                    newAnwers,
                    journey
                  ).fold({ e =>
                    logger.warn("Could not update draft return", e)
                    errorHandler.errorResult()
                  }, _ => Redirect(routes.RepresenteeController.checkYourAnswers()))
```

Reply:
Done.